octavia_f5/controller/worker/controller_worker.py

```diff
@@ -99,17 +99,24 @@ def pending_sync(self):
             # bind to loadbalancer if scheduled to this host
             if CONF.host == self.network_driver.get_scheduled_host(lb.vip.port_id):
                 self.ensure_amphora_exists(lb.id)
-                lbs.add(lb)
+                if not lb in lbs: # deduplicate
+                    lbs.append(lb)
 
-        lbs.update(self._loadbalancer_repo.get_all_from_host(
-            db_apis.get_session(),
-            provisioning_status=lib_consts.PENDING_UPDATE))
+        pending_lbs = self._loadbalancer_repo.get_all_from_host(
+            db_apis.get_session(), provisioning_status=lib_consts.PENDING_UPDATE)
+        for lb in pending_lbs:
+            if not lb in lbs:
+                lbs.append(lb)
 
         pools = self._pool_repo.get_pending_from_host(db_apis.get_session())
-        lbs.update([pool.load_balancer for pool in pools])
+        for lb in [pool.load_balancer for pool in pools]:
+            if not lb in lbs:
+                lbs.append(lb)
 
         l7policies = self._l7policy_repo.get_pending_from_host(db_apis.get_session())
-        lbs.update([l7policy.listener.load_balancer for l7policy in l7policies])
+        for lb in [l7policy.listener.load_balancer for l7policy in l7policies]:
+            if not lb in lbs:
+                lbs.append(lb)
```

Comment:
Thanks for looking into it, although I am not satisfied with the solution since we could still redundandly update multiple time if two or more LBs are pending in a single tenant (same network_id). Please look into my PR, thanks.